pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/SimpleLoadManagerImpl.java

```diff
@@ -940,8 +941,8 @@ private synchronized ResourceUnit findBrokerForPlacement(Multimap<Long, Resource
         }
     }
 
-    public ResourceUnit getLeastLoaded(ServiceUnitId serviceUnit) throws Exception {
-        return getLeastLoadedBroker(serviceUnit, getAvailableBrokers(serviceUnit));
+    public Optional<ResourceUnit> getLeastLoaded(ServiceUnitId serviceUnit) throws Exception {
+        return Optional.of(getLeastLoadedBroker(serviceUnit, getAvailableBrokers(serviceUnit)));
```

Comment:
can `getLeastLoadedBroker(..)` return null broker? in that case should we use `Optional.ofNullable(..)` ?

Reply:
Sounds good, I didn't want to enter in the SimpleLoadManager but that makes sense.